### PR TITLE
[RDY] Allow dmg files as disk images for TH data

### PR DESCRIPTION
**Pull request by tobylane**
**Describe what the proposed change does**
- Generalises iso to disk images, currently .iso and .dmg.

### CODE REVIEW

**Comment by lewri** [comment]
Are there any knock on behaviours for accessing our files via DMG e.g. is it restricted to the same issues as being unable to play videos?

**Comment by tobylane** [comment]
Yes, this gives the dmg to the same iso_fs, so it will have that limitation.

File: CorsixTH/Lua/filesystem.lua [hunk 1]
```diff
@@ -75,13 +75,14 @@ function FileSystem:_enumerate()
...
+-- If the path is a disk image then set the provider. If the path is a directory
 -- then set the pysical_path and populate the files and sub_dirs.
 --
 --!param physical_path (string) a path on the filesystem to either a directory
--- or theme hospital ISO file.
+-- or theme hospital ISO or DMG file.
 function FileSystem:setRoot(physical_path)
-  if physical_path:match"%.[iI][sS][oO]$" or physical_path:match"%.[iI][sS][oO]9660$" then
+  local path = physical_path:lower()
+  if path:match("%.iso$") or path:match("%.iso9660$") or path:match("%.dmg$") then
```
Comment (driverpt) [inline CorsixTH/Lua/filesystem.lua:85]: Is this Case-Insensitive ?

**Review by driverpt — CHANGES_REQUESTED** [review]
`.dmg` is Mac-specific. You should add some `if os.type =='darwin'` check to prevent issues.

**Comment by tobylane** [comment]
In this case it's just another name for the same format, a name that the native tool uses. I could transfer my disk image to Windows and not need to rename it.

**Comment by TheCycoONE** [comment]
dmg's can be a variety of formats, and the only format we support is ISO9660 so I think the isIso / ISO comments can remain as they are; or maybe explicitly `isISO9660`.

**Comment by tobylane** [comment]
Reverted those parts and fixed a bug.

File: CorsixTH/Lua/filesystem.lua (hunk 1 above)
Comment (TheCycoONE) [reply CorsixTH/Lua/filesystem.lua:85]: @driverpt There's a tolower just above now. This path is only being used for the match check.

File: CorsixTH/Src/th_lua_iso.cpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ int l_isofs_set_root(lua_State* L) {
   const char* path = luaL_checkstring(L, 2);
   if (pSelf->initialise(path)) {
     lua_pushvalue(L, 2);
-    luaT_setenvfield(L, 1, "file");
```
Comment (TheCycoONE) [inline CorsixTH/Src/th_lua_iso.cpp:47]: good catch

**Comment by TheCycoONE** [comment]
LGTM, ready when you are.

File: CorsixTH/Lua/filesystem.lua [hunk 3]
```diff
@@ -76,12 +76,13 @@ end
...
 -- If the path is an ISO then set the provider. If the path is a directory
--- then set the pysical_path and populate the files and sub_dirs.
+-- then set the physical_path and populate the files and sub_dirs.
 --
 --!param physical_path (string) a path on the filesystem to either a directory
 -- or theme hospital ISO file.
 function FileSystem:setRoot(physical_path)
-  if physical_path:match"%.[iI][sS][oO]$" or physical_path:match"%.[iI][sS][oO]9660$" then
+  local path = physical_path:lower()
+  if path:match("%.iso$") or path:match("%.iso9660$") or path:match("%.dmg$") then
```
Comment (TheCycoONE) [inline CorsixTH/Lua/filesystem.lua:85]: Any easy way to share this check with the filesystem check for the same thing? (I like the way it was done there a little better, if you had to choose implementation)

**Comment by tobylane** [comment]
Both now use FileSystem:isIso. Ready.

**Pull request merged**